test/libsolidity/SemanticTest.cpp

```diff
@@ -340,13 +447,30 @@ void SemanticTest::printUpdatedSettings(ostream& _stream, string const& _linePre
 
 void SemanticTest::parseExpectations(istream& _stream)
 {
-	TestFileParser parser{_stream};
+	TestFileParser parser{_stream, &this->m_builtins};
 	auto functionCalls = parser.parseFunctionCalls(m_lineOffset);
 	std::move(functionCalls.begin(), functionCalls.end(), back_inserter(m_tests));
 }
 
-bool SemanticTest::deploy(string const& _contractName, u256 const& _value, bytes const& _arguments, map<string, solidity::test::Address> const& _libraries)
+bool SemanticTest::deploy(
+	string const& _contractName,
+	u256 const& _value,
+	bytes const& _arguments,
+	map<string, solidity::test::Address> const& _libraries)
 {
 	auto output = compileAndRunWithoutCheck(m_sources.sources, _value, _contractName, _arguments, _libraries);
 	return !output.empty() && m_transactionSuccessful;
 }
+
+std::optional<bytes> SemanticTest::builtinSmokeTest(FunctionCall const& call)
```

Comment:
I think I should move this to the `SmokeBehaviour`.

Reply:
Probably there is no real need to have `SmokeHooks` (`SmokeBehaviour`). I thought about them like examples, but we will have real test hook examples (gas expactation & isoltest events) soon. These can be used to find out how the test hooks work.